api/src/main/java/io/strimzi/api/kafka/model/common/authentication/KafkaClientAuthentication.java

```diff
@@ -35,10 +36,11 @@ public abstract class KafkaClientAuthentication implements UnknownPropertyPreser
     private Map<String, Object> additionalProperties;
 
     @Description("Authentication type. " +
-            "Currently the supported types are `tls`, `scram-sha-256`, `scram-sha-512`, `plain`, and 'oauth'. " +
+            "Currently the supported types are `tls`, `scram-sha-256`, `scram-sha-512`, `plain`, 'oauth', and 'serviceaccount-oauth'. " +
             "`scram-sha-256` and `scram-sha-512` types use SASL SCRAM-SHA-256 and SASL SCRAM-SHA-512 Authentication, respectively. " +
             "`plain` type uses SASL PLAIN Authentication. " +
             "`oauth` type uses SASL OAUTHBEARER Authentication. " +
+            "`serviceaccount-oauth` type uses SASL OAUTHBEARER Authentication. " +
```

Comment:
Maybe we could add something like `uses SASL OAUTHBEARER Authentication in combination with Kubernetes ServiceAccounts`?

Reply:
Yes, sounds good.

Reply:
Updated the text.